acme/acme/crypto_util.py

```diff
@@ -442,15 +442,12 @@ def make_self_signed_cert(private_key: CertificateIssuerPrivateKeyTypes,
             critical=False
         )
 
-    not_before_datetime = datetime.fromtimestamp(
-        0 if not_before is None else not_before
-    )
-    builder = builder.not_valid_before(not_before_datetime)
-    validity_duration = timedelta(seconds=validity)
     if not_before is None:
-        not_valid_after = _now() + validity_duration
+        not_before_datetime = _now()
     else:
-        not_valid_after = not_before_datetime + validity_duration
+        not_before_datetime = datetime.fromtimestamp(not_before)
```

Comment:
This doesn't match the behavior of `not_before` on pyOpenSSL. And to be clear, and I think this behavior is better!

```
>>> c.gmtime_adj_notAfter(10)
>>> c.get_notAfter()
b'20250107211013Z'
```

example of what it looks like on pyOpenSSL -- you can see it sets it relative to "now".

Given your behavior is 100x saner, have you verified all the callers do the right thing?

Reply:
as it turns out, none of our internal callers use this parameter, so i think we're good!

Reply:
lol. lmao. maybe just make the function take a datetime instead, and avoid this comedy?

Reply:
Besides this, the PR looks substantively fine to me! Thanks!

Reply:
that's a great idea actually. i guess the only wrinkle with that is handling naive datetimes vs timezone aware ones, the docstring can just specify naive ones are assumed to be UTC